corehq/apps/translations/app_translations/upload_app.py

```diff
@@ -38,30 +39,71 @@
     get_single_worksheet,
 )
 
+from corehq.util.files import read_workbook_content_as_file
 
-def validate_bulk_app_translation_upload(app, workbook, email, lang_to_compare):
+from CommcareTranslationChecker import validate_workbook
+from CommcareTranslationChecker.exceptions import FatalError
+
+
+def run_translation_checker(file_obj):
+    translation_checker_messages = []
+    result_wb = None
+    try:
+        result_wb = validate_workbook(file_obj, translation_checker_messages)
+    except FatalError as e:
+        translation_checker_messages.append(
+            _("Workbook check failed to finish due to the following error : %s" % e))
+    return translation_checker_messages, result_wb
+
+
+def validate_bulk_app_translation_upload(app, workbook, email, lang_to_compare, file_obj):
     from corehq.apps.translations.validator import UploadedTranslationsValidator
     msgs = UploadedTranslationsValidator(app, workbook, lang_to_compare).compare()
-    if msgs:
-        _email_app_translations_discrepancies(msgs, email, app.name)
+    checker_messages, result_wb = run_translation_checker(file_obj)
+    if msgs or checker_messages:
+        _email_app_translations_discrepancies(msgs, checker_messages, email, app.name, result_wb)
         return [(messages.error, _("Issues found. You should receive an email shortly."))]
     else:
         return [(messages.success, _("No issues found."))]
 
 
-def _email_app_translations_discrepancies(msgs, email, app_name):
-    html_content = ghdiff.default_css
-    for sheet_name, msg in msgs.items():
-        html_content += "<strong>{}</strong>".format(sheet_name) + msg
+def _email_app_translations_discrepancies(msgs, checker_messages, email, app_name, result_wb):
+    """
+    :param msgs: messages for app translation discrepancies
+    :param checker_messages: messages for issues found by translation checker
+    :param email: email to
+    :param app_name: name of the application
+    :param result_wb: result wb of translation checker to attach with the email
+    """
+    def _form_email_content(msgs, checker_messages):
```

Comment:
these two methods dont need to set as private since they are already within the scope of this method, so the `_` is not really needed.